crates/block-producer/src/block_builder/mod.rs

```diff
@@ -223,20 +246,88 @@ impl BlockBuilder {
         &self,
         mempool: &SharedMempool,
         built_block: ProvenBlock,
-    ) -> Result<(), BuildBlockError> {
+    ) -> Result<StateDelta, BuildBlockError> {
         self.store
             .apply_block(&built_block)
             .await
             .map_err(BuildBlockError::StoreApplyBlockFailed)?;
 
-        mempool.lock().await.commit_block();
-
-        Ok(())
+        let reverted_transactions = mempool.lock().await.commit_block();
+        let committed_transactions = built_block
+            .transactions()
+            .as_slice()
+            .iter()
+            .map(TransactionHeader::id)
+            .collect();
+        let committed_network_notes = built_block
+            .output_notes()
+            .filter_map(|(_idx, note)| match note {
+                OutputNote::Full(inner)
+                    if inner.metadata().tag().execution_mode() == NoteExecutionMode::Network =>
+                {
+                    Some(inner.clone())
+                },
+                _ => None,
+            })
+            .collect();
```

Comment:
nit: I would maybe create a small helper function which takes a reference to a block and returns a vector of network notes from that block.